tests/rptest/tests/adjacent_segment_merging_test.py

```diff
@@ -48,6 +48,10 @@ def __init__(self, test_context):
             cloud_storage_idle_timeout_ms=200,
             cloud_storage_segment_size_target=1024 * 1024 * 10,
             cloud_storage_segment_size_min=1024 * 1024 * 8,
+            # Disable compacted segment reupload to allow adjacent
+            # merger to do it's job. Otherwise the reupload may
+            # merge and reupload the whole range.
+            cloud_storage_enable_compacted_topic_reupload=False,
```

Comment:
I got confused reading this. From the previous commit, my understanding was that the adjacent segment merger will only merge segments that have already been through the compacted reuplod (it the topic has compaction enabled). The fact that we disable it here, clashes with my understanding.

Reply:
I'll change this. Currently, it can work before compacted reuploader but looks like this is not a good idea.